### DISCUSSION TITLE
Textarea loses focus when reactive

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#1198] [Help]
I have set up basic editor which provides two tabs - one for rich editing and one for HTML. They share the same field - 'content' and I have made them reactive so that each will update the other:

```
Tabs::make('Content')
    ->tabs([
        Tab::make('Text')
            ->schema([
                RichEditor::make('content')
                    ->fileAttachmentsDisk('public')
                    ->fileAttachmentsDirectory('media')
                    ->reactive()
                    ->afterStateUpdated(fn($state, callable $set) => $set('content', $state)),
            ]),
        Tab::make('HTML')
            ->schema([
                Textarea::make('content')
                    ->rows(25)
                    ->reactive()
                    ->afterStateUpdated(fn($state, callable $set) => $set('content', $state))
            ])
    ]),
```
When changing the content of the RichEditor, everything works fine and switching to the HTML tab shows the changes. However when editing the TextArea, the changes are reflected in the RichEditor but the control loses focus after any change meaning you have to click back into it after entering one or two characters to continue editing.

As suggested, I may be approaching this wrongly so any suggestions appreciated!

**Comment by danharrin** [comment]
You shouldn't need the `afterStateUpdated()` here - since Livewire does that anyway. Right?

**Reply by titantwentyone** [reply]
Thanks Dan. Removing `afterStateUpdated` produces the same reactive effect but the lost focus is still an issue.

Interestingly, I took these fields out of the the tabs and placed them side by side. When I enter in the TextArea, focus moves to the RichEditor and the cursor is placed there. The order in which they are rendered also doesn't seem to make a difference.

I did wonder if this was because there were now two elements on the page with an id of "data.content". I attempted to do the following with my TextArea:

```
Textarea::make('content')
    ->rows(25)
    ->reactive()
    ->extraAttributes([
        'id' => 'content2'
    ]),
```
But the `id` field doesn't take hold. Even if I could do this, I'm guessing the framework generates code to move focus to 'data.content' so this may be moot.

The base issue, I suspect, is the fact that I am trying to update one field with two controls. Is there another way to approach this?

**Reply by titantwentyone** [reply]
I'm really not sure if this helps (apologies if confusing the matter) but I did try a modified approach shown below to get rid of the potential isse of two elements with the same id on the page however this results in the same behaviour - TextArea loses focus and passes to RichEditor. My idea was to just have one "content" control (which would ultimately be hidden) and provide two additional controls which update "content" - "content_html" and "content_rich":

 ```
RichEditor::make('content_rich')
    ->fileAttachmentsDisk('public')
    ->fileAttachmentsDirectory('/media')
    ->reactive()
    ->afterStateUpdated(function($state, callable $set) {
        $set('content', $state);
        $set('content_html', $state);
    }),
Textarea::make('content_html')
    ->rows(25)
    ->reactive()
    ->afterStateUpdated(function($state, callable $set) {
        $set('content', $state);
        $set('content_rich', $state);
    }),
Textarea::make('content')
    ->afterStateHydrated(function($state, callable $set) {
        $set('content_rich', $state);
        $set('content_html', $state);
    })
    ->rows(25)
```

**Reply by danharrin** [reply]
Hey, you can give two fields different IDs using the `id()` method - try that. I'd stay away from `afterStateUpdated()` etc here.

**Reply by titantwentyone** [reply]
No joy with first example I provided (minus `afterStateUpdated`). Still same behaviour. Assume this woudn't matter with the second as the fields are named differently anyway.

This is in a branch for upgrading to Filament v2 so more than happy to try out any fixes/suggestions if needed. Long term, I'll be moving content over to the new Builder control but have a lot of content that will need this in the interim while that migration occurs. Many thanks for your help!
